content-src/components/NewTabPage/NewTabPage.js

```diff
@@ -17,27 +17,33 @@ const PAGE_NAME = "NEW_TAB";
 const NewTabPage = React.createClass({
   getInitialState() {
     return {
-      showSettingsMenu: false
+      showSettingsMenu: false,
+      renderedOnce: false
     };
   },
-  resetBlockList() {
+  toggleRecommendation() {
     this.props.dispatch(actions.NotifyEvent({
-      event: "UNBLOCK_ALL",
+      event: "TOGGLE_RECOMMENDATION",
       page: PAGE_NAME
     }));
-    this.props.dispatch(actions.NotifyUnblockAll());
+    this.props.dispatch(actions.NotifyToggleRecommendations());
+    this.props.dispatch(actions.RequestHighlightsLinks());
   },
   componentDidMount() {
     document.title = "New Tab";
     setFavicon("newtab-icon.svg");
   },
   componentDidUpdate() {
-    if (this.props.isReady) {
+    if (this.props.isReady && !this.state.renderedOnce) {
       this.props.dispatch(actions.NotifyPerf("NEWTAB_RENDER"));
+      this.setState({renderedOnce: true});
```

Comment:
Oh actually, I would guess isReady is not true initially, so that's why we need to wait on an update.